### PR TITLE
Deprecate partial attribute

### PR DESCRIPTION
**Pull request by alshakero**
Changed all `partial` property and attribute usages in tests in favor of `view-model` and viewModel. And added tests for the warnings.

### CODE REVIEW

File: starcounter-include.html [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 <script>
     (function() {
+        function warnAboutDeperecatedPartial() {
+            console.warn('`partial` attribute and property are deprecatrd from `starcounter-include` in fovour of (viewModel property or view-model attribute), they will soon be no longer be supported');
```
Comment (tomalec) [inline starcounter-include.html]: "deprecatrd" -> "deprecated", fovour ->favour

File: starcounter-include.html [hunk 2]
```diff
@@ -30,6 +33,11 @@
                 this.defaultComposition = null;
                 var partialId = this.partialId;
                 var partial = this.viewModel || this.partial || undefined;
```
Comment (tomalec) [inline starcounter-include.html]: I think we should use `viewModel` instead of `partial` internally as wel..

File: starcounter-include.html [hunk 3]
```diff
@@ -63,7 +74,9 @@
                     },
                     viewModel: {
                         set: function(newValue) {
+                            this.settingPartialFromModel = true;
```
Comment (tomalec) [inline starcounter-include.html]: If we will use `viewModel` internall (which we should in my opinion), that kind of magic will not be needed. Unless you do this just to reflect the old `partial` values set to `viewModel`

File: starcounter-include.html [hunk 4]
```diff
@@ -153,9 +166,16 @@
          * Set partial property if partial or viewm-model attribute was changed
          */
         StarcounterIncludePrototype.attributeChangedCallback = function(name, oldVal, newVal) {
-            if (name === "partial" || name === "view-model") {
+            if (name === "view-model") {
+                this.settingPartialFromModel = true;
+                this.partial = partialAttributeToProperty(this, newVal);
```
Comment (tomalec) [inline starcounter-include.html]: We should use `.viewModel` internally

**Review by tomalec — CHANGES_REQUESTED** [review]

**Comment by alshakero** [comment]
Done.

File: .vs/ProjectSettings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (tomalec) [inline .vs/ProjectSettings.json]: Please add those files to gitignore

**Review by tomalec — CHANGES_REQUESTED** [review]

**Review by warpech — APPROVED** [review]
Lgtm

**Comment by warpech** [comment]
I was asked to merge by @alshakero, even though the status of @tomalec review is still red. We believe @tomalec would change to green now, but he can't since he is in Tokio now.

 Merging.

**Pull request merged**